src/Features/Core/Portable/ImplementAbstractClass/ImplementAbstractClassData.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.ImplementType;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ImplementAbstractClass
+{
+    internal class ImplementAbstractClassData
+    {
+        private readonly Document _document;
+        private readonly SyntaxNode _classNode;
+        private readonly ImmutableArray<(INamedTypeSymbol type, ImmutableArray<ISymbol> members)> _unimplementedMembers;
+
+        public readonly INamedTypeSymbol ClassType;
+
+        public ImplementAbstractClassData(
+            Document document, SyntaxNode classNode, INamedTypeSymbol classType,
+            ImmutableArray<(INamedTypeSymbol type, ImmutableArray<ISymbol> members)> unimplementedMembers)
+        {
+            _document = document;
+            _classNode = classNode;
+            ClassType = classType;
+            _unimplementedMembers = unimplementedMembers;
+        }
+
+        public static async Task<ImplementAbstractClassData?> TryGetDataAsync(
```

Comment:
nit: TryGetFoo makes me think bool returning. Is there another naming scheme we can use here?

Reply:
TryXXX is very common in teh codebase to mean "can return null".  We could consider if that's necessary nowadays for NRT enabled code though.  Then it can just be GetDataAsync with the clear annotation explaining what's going on.  I'm not sure if we should prefer consistency though up until we have NRT enabled everywhere.